x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -147,7 +147,14 @@ export class Plugin implements ISecuritySolutionPlugin {
     const eventLogService = plugins.eventLog;
     registerEventLogProvider(eventLogService);
 
-    const requestContextFactory = new RequestContextFactory({ config, logger, core, plugins });
+    const requestContextFactory = new RequestContextFactory({
+      config,
+      logger,
+      core,
+      plugins,
+      endpointAppContextService: this.endpointAppContextService,
```

Comment:
These services are already exposed on the `endpointAppContextService` a few lines below:

https://github.com/elastic/kibana/blob/d3864e136154b0ef869ea2e4d2e9c29f14033c3e/x-pack/plugins/security_solution/server/plugin.ts#L164-L169

So another option would be just wiring up the `endpointContext` through `initRoutes` and passing the existing context to the route ala:

```
readRulesRoute(router, logger, endpointAppContext);
```

---
It's been a bit since I've been in our plugin setup code, so not sure of all the trade-offs here, but wanted to mention since this was the route (har har 😅) I took over in my initial [spike branch](https://github.com/elastic/kibana/compare/main...spong:spike/rule-compatibility-checker?expand=1).